src/components/Search.js

```diff
@@ -6,26 +6,29 @@ class Search extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      // default placesholders while page is loading
-      location: 'Enter Location',
-      date: 'mm/dd/yyyy',
+      location: 'Enter a location',
     };
   }
 
-  componentDidMount() {
-    // sets placeholder values of the date & location fields to curr loc and curr date
-    const today = getdate('mm/dd/yyyy');
-    const _this = this;
+  componentWillMount() {
+    // sets current location for on page load
     getloc((loc) => {
-      _this.setState(
-        {
-          location: loc,
-          date: today,
-        }
-      );
+      this.setState({
+        location: loc,
+      });
     });
   }
 
+  setLocation(event) {
```

Comment:
Interesting choice; I think I see a way this code could be refactored to be more DRY. Currently two places in the code set the state of location. This one handles both the logic of setting the state of location, and it also interprets an on-click event.

It might be better to separate these concerns out with a method that sets the state of location ( perhaps called setLocation ). It might be useful to have another method ( handleClick perhaps? ) to handle the logic of what to do with that click event ( calling setLocation ) and also re-use setLocation on line 17.